### PR TITLE
Add ignore: ["next-sibling"] to selector-max-type

### LINKED ISSUE #3757: Add ignore: ["next-sibling"] to selector-max-type
> What is the problem you're trying to solve?

Need Combinators as ignore option for "selector-max-type" rule.

> What solution would you like to see?

A new  option should ignore "Subsequent-sibling combinator " and "Next-sibling combinator"
 in "selector-max-type" rule.

The rule should ignore below css
``` #userSelVal + span{}```
``` #userSelVal > span{}```

### PR DESCRIPTION
**Pull request by iliyaZelenko**
> Which issue, if any, is this issue related to?

Closes #3757

> Is there anything in the PR that needs further explanation?

"No, it's self explanatory."

### CODE REVIEW

File: lib/rules/selector-max-type/__tests__/index.js [hunk 1]
```diff
@@ -549,3 +549,82 @@ testRule(rule, {
...
+    {
+      code: ".foo + a + b {}"
+    },
+    {
+      code: ".foo { .bar + a {} }"
+    },
+    {
+      code: ".foo, .bar + baz {}"
+    },
+    {
```
Comment (jeddy3) [inline lib/rules/selector-max-type/__tests__/index.js]: I think we can remove this test. It doesn't add much and I believe the trailing comma after the closing brace makes it invalid CSS.

File: lib/rules/selector-max-type/__tests__/index.js [hunk 2]
```diff
@@ -549,3 +549,82 @@ testRule(rule, {
     }
   ]
 });
+
+testRule(rule, {
+  ruleName,
+  config: [0, { ignore: ["next-sibling"] }],
+  skipBasicChecks: true,
+  syntax: "scss",
```
Comment (jeddy3) [inline lib/rules/selector-max-type/__tests__/index.js]: I don't think we need to test non-standard syntax here:

```suggestion
```

File: lib/rules/selector-max-type/__tests__/index.js [hunk 3]
```diff
@@ -549,3 +549,82 @@ testRule(rule, {
...
+      code: ".foo + a {}"
+    },
+    {
+      code: ".foo, .bar + a {}"
+    },
+    {
+      code: ".foo + a + b {}"
+    },
+    {
+      code: ".foo { .bar + a {} }"
```
Comment (jeddy3) [inline lib/rules/selector-max-type/__tests__/index.js]: Let's use standard syntax here by adding the _nesting selector_:

```suggestion
      code: ".foo { & .bar + a {} }"
```

File: lib/rules/selector-max-type/__tests__/index.js [hunk 4]
```diff
@@ -549,3 +549,82 @@ testRule(rule, {
...
+      column: 1
+    },
+    {
+      code: ".foo + bar, baz {}",
+      message: messages.expected("baz", 0),
+      line: 1,
+      column: 13
+    },
+    {
+      code: ".foo + .bar { baz {} }",
```
Comment (jeddy3) [inline lib/rules/selector-max-type/__tests__/index.js]: Let's use standard syntax here by adding the nesting selector:

```suggestion
      code: ".foo + .bar { & baz {} }",
```

File: lib/rules/selector-max-type/__tests__/index.js [hunk 5]
```diff
@@ -549,3 +549,82 @@ testRule(rule, {
...
+      message: messages.expected(".foo + .bar baz", 0),
+      line: 1,
+      column: 15
+    },
+    {
+      code: ".foo + .bar { .baz + .foo { foo + bar {} } }",
```
Comment (jeddy3) [inline lib/rules/selector-max-type/__tests__/index.js]: Let's use standard syntax here by adding the nesting selector:

```suggestion
      code: ".foo + .bar { & .baz + .foo { & foo + bar {} } }",
```

File: lib/rules/selector-max-type/index.js [hunk 6]
```diff
@@ -170,7 +175,11 @@ function isDescendantCombinator(node) {
 function isChildCombinator(node) {
   if (!node) return false;
 
-  return isCombinator(node) && node.value.includes(">");
+  return isCombinator(node) && node.value === ">";
```
Comment (jeddy3) [inline lib/rules/selector-max-type/index.js]: I wonder why this was `.includes` before? 

Do we know if a combinator node can contain whitespace e.g. `/n>`? 

Can you added a test for this please e.g. the following should have no violations:

```
code: ".foo \n+\n a {}"
```

Comment (iliyaZelenko) [reply lib/rules/selector-max-type/index.js]: Added, it seems to work.

File: lib/rules/selector-max-type/README.md [hunk 7]
```diff
@@ -119,6 +119,24 @@ The following patterns are *not* considered violations:
 #bar div span a {}
 ```
 
+#### "next-sibling"
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md]:
```suggestion
#### `"next-sibling"`
```

File: lib/rules/selector-max-type/README.md [hunk 8]
```diff
@@ -119,6 +119,24 @@ The following patterns are *not* considered violations:
...
+
+```css
+div a + span {}
+```
+
+```css
+#bar + div + span + a + span {}
+```
+
+
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md]:
```suggestion
```

File: lib/rules/selector-max-type/README.md [hunk 9]
```diff
@@ -119,6 +119,24 @@ The following patterns are *not* considered violations:
...
+The following patterns are *not* considered violations:
+
+```css
+div a + span {}
+```
+
+```css
+#bar + div + span + a + span {}
+```
+
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md]:
```suggestion
```

**Comment by jeddy3** [comment]
Does anyone have the time to provide a second review on this?

**Comment by jeddy3** [comment]
@iliyaZelenko Sorry to trouble you once last time, but would you be able to resolve the conflicts so we can merge this?

**Comment by jeddy3** [comment]
> The tests did not pass, I will try to solve it in the near future.

I think we're simply missing the "next-sibling" option in the validator. It looks like it got removed when resolving the conflicts. Adding it back in should see the tests passing again. See: https://github.com/stylelint/stylelint/blob/2965800f8eccf0dcfa75ccf9f1fb499e257e21a7/lib/rules/selector-max-type/index.js#L38

**Comment by iliyaZelenko** [comment]
@jeddy3 You are right, thank you, did not notice. 

Everything is working.

**Comment by jeddy3** [comment]
Changelog entry added:

> -   Added: `ignore: ["next-sibling"]` to `selector-max-type` ([#3832](https://github.com/stylelint/stylelint/pull/3832)).

**Pull request merged**